src/dbcp/transform/lbnl_iso_queue.py

```diff
@@ -108,31 +114,39 @@ def _harmonize_interconnection_status_lbnl(statuses: pd.Series) -> pd.Series:
         "Withdrawn",
     }
     out = statuses.replace(mapping)
-    bad = out.loc[~out.isin(allowed_statuses)]
-    assert len(bad) == 0, f"Unknown interconnection status(es): {bad.value_counts()}"
+    bad = out.loc[~out.isin(allowed_statuses) & out.notna()]
+    assert len(bad) == 0, f"Unknown interconnection status(es):\n{bad.value_counts()}"
     return out
 
 
-def active_iso_queue_projects(active_projects: pd.DataFrame) -> pd.DataFrame:
-    """Transform active iso queue data."""
+def _clean_all_iso_projects(projects: pd.DataFrame) -> pd.DataFrame:
+    """Transform active, operational and withdrawn iso queue projects."""
     rename_dict = {
         "state": "raw_state_name",
         "county": "raw_county_name",
     }
-    active_projects["project_id"] = np.arange(len(active_projects), dtype=np.int32)
-    active_projects = active_projects.rename(columns=rename_dict)  # copy
-    active_projects.loc[
+    projects["project_id"] = np.arange(len(projects), dtype=np.int32)
+    projects = projects.rename(columns=rename_dict)  # copy
+    projects.loc[
         :, "interconnection_status_lbnl"
     ] = _harmonize_interconnection_status_lbnl(
-        active_projects.loc[:, "interconnection_status_lbnl"]
+        projects.loc[:, "interconnection_status_lbnl"]
+    )
+    parse_date_columns(projects)
+    # rename date_withdrawn to withdrawn_date and date_operational to actual_completion_date
+    projects.rename(
+        columns={
+            "date_withdrawn_raw": "withdrawn_date_raw",
+            "date_operational_raw": "actual_completion_date_raw",
+            "date_withdrawn": "withdrawn_date",
+            "date_operational": "actual_completion_date",
+        },
+        inplace=True,
     )
-    # drop irrelevant columns (structurally all nan due to 'active' filter)
-    active_projects.drop(columns=["date_withdrawn", "date_operational"], inplace=True)
-    parse_date_columns(active_projects)
     # deduplicate
-    pre_dedupe = len(active_projects)
-    active_projects = deduplicate_active_projects(
-        active_projects,
+    pre_dedupe = len(projects)
+    projects = deduplicate_active_projects(
```

Comment:
@TrentonBush do you think this deduplication algorithm work on operation and withdrawn projects?